db/change_listener.go

```diff
@@ -95,8 +95,8 @@ func (listener *changeListener) ProcessFeedEvent(event sgbucket.FeedEvent) bool
 	requiresCheckpointPersistence := true
 	if event.Opcode == sgbucket.FeedOpMutation || event.Opcode == sgbucket.FeedOpDeletion {
 		key := string(event.Key)
-		if strings.HasPrefix(key, auth.UserKeyPrefix) ||
-			strings.HasPrefix(key, auth.RoleKeyPrefix) { // SG users and roles
+		if (strings.HasPrefix(key, auth.UserKeyPrefix) ||
+			strings.HasPrefix(key, auth.RoleKeyPrefix)) && event.Opcode == sgbucket.FeedOpMutation { // SG users and roles
```

Comment:
Changed, I ran these commands manually and the resulting log entries:

`curl -X POST http://localhost:4985/db/_user/ -H "Content-Type: application/json" -d '{"name":"SyncGW1C3","admin_roles":["level-2"],"password":"password","admin_channels":["SyncGW1C3"]}'`

```
2017-08-22T17:35:40.224+01:00 HTTP:  #001: POST /db/_user/  (ADMIN)
2017-08-22T17:35:40.279+01:00 Access: Computed channels for "SyncGW1C3": !:1
2017-08-22T17:35:40.283+01:00 Access: Computed roles for "SyncGW1C3": 
2017-08-22T17:35:40.370+01:00 Auth: Saved _sync:user:SyncGW1C3: &{{SyncGW1C3 SyncGW1C3:24  %!s(uint64=24)  %!s(*uint16=<nil>)} { %!s(bool=false) $2a$10$kI3Q8pOSR149TJ4emnIK6eLC7iUZ2vnth7YqavNHElYm1fmrBBLru <nil> level-2:24  []} %!s(*auth.Authenticator=&{0xc4201baf00 0xc42015cf00}) []}
2017-08-22T17:35:40.371+01:00 HTTP+: #001:     --> 201 Created  (146.9 ms)
2017-08-22T17:35:40.371+01:00 Cache: Received #24 ("_user/SyncGW1C3")
2017-08-22T17:35:40.371+01:00 Cache: Initialized cache for channel "*" with options: &{ChannelCacheMinLength:50 ChannelCacheMaxLength:500 ChannelCacheAge:1m0s}
2017-08-22T17:35:40.371+01:00 Cache:     #24 ==> channel "*"
2017-08-22T17:35:40.371+01:00 Changes+: Notifying that "bucket-1" changed (keys="{*}") count=2
2017-08-22T17:35:40.371+01:00 Changes+: Notifying that "bucket-1" changed (keys="{_sync:user:SyncGW1C3}") count=3
```

`curl -X DELETE http://localhost:4985/db/_user/SyncGW1C3`

```
2017-08-22T17:35:44.238+01:00 HTTP:  #002: DELETE /db/_user/SyncGW1C3  (ADMIN)
2017-08-22T17:35:44.275+01:00 Access: Computed channels for "SyncGW1C3": !:1,SyncGW1C3:24
2017-08-22T17:35:44.278+01:00 Access: Computed roles for "SyncGW1C3": level-2:24
2017-08-22T17:35:44.279+01:00 HTTP+: #002:     --> 200   (41.0 ms)
2017-08-22T17:35:44.279+01:00 Changes+: Notifying that "bucket-1" changed (keys="{_sync:user:SyncGW1C3}") count=4
```

```
curl -X GET http://localhost:4985/db/_changes
{"results":[
{"seq":5,"id":"doc1","changes":[{"rev":"2-b930e9ed5cbc1165d7c14d3c5834b4c4"}]}
,{"seq":12,"id":"doc100","deleted":true,"changes":[{"rev":"7-0f81f093bb3d28686a2af83501244623"}]}
,{"seq":24,"id":"_user/SyncGW1C3","changes":[{"rev":""}]}
],
"last_seq":"24"}
```

```
2017-08-22T17:35:56.813+01:00 HTTP:  #003: GET /db/_changes  (ADMIN)
2017-08-22T17:35:56.813+01:00 Changes+: Int sequence multi changes feed...
2017-08-22T17:35:56.813+01:00 Changes: MultiChangesFeed(channels: {*}, options: {Since:0 Limit:0 Conflicts:false IncludeDocs:false Wait:false Continuous:false Terminator:0xc4202b3440 HeartbeatMs:0 TimeoutMs:300000 ActiveOnly:false}) ... 
2017-08-22T17:35:56.813+01:00 Changes+: MultiChangesFeed: channels expand to "" ... 
2017-08-22T17:35:56.813+01:00 Cache: getCachedChanges("*", 0) --> 1 changes valid from #24
2017-08-22T17:35:56.813+01:00 Cache:   Querying 'channels' view for "*" (start=#1, end=#24, limit=0)
2017-08-22T17:35:56.852+01:00 Cache:     Got 2 rows from view for "*": #5 ... #12
2017-08-22T17:35:56.852+01:00 Cache: GetChangesInChannel("*") --> 3 rows
2017-08-22T17:35:56.852+01:00 Changes+: [changesFeed] Found 3 changes for channel *
2017-08-22T17:35:56.852+01:00 Changes+: Channel feed processing seq:5 in channel * 
2017-08-22T17:35:56.852+01:00 Changes+: Channel feed processing seq:12 in channel * 
2017-08-22T17:35:56.852+01:00 Changes+: Channel feed processing seq:24 in channel * 
2017-08-22T17:35:56.852+01:00 Changes+: MultiChangesFeed sending {Seq:5, ID:doc1, Changes:[map[rev:2-b930e9ed5cbc1165d7c14d3c5834b4c4]]} 
2017-08-22T17:35:56.852+01:00 Changes+: MultiChangesFeed sending {Seq:12, ID:doc100, Changes:[map[rev:7-0f81f093bb3d28686a2af83501244623]], Deleted:true} 
2017-08-22T17:35:56.852+01:00 Changes+: MultiChangesFeed sending {Seq:24, ID:_user/SyncGW1C3, Changes:[map[rev:]]} 
2017-08-22T17:35:56.852+01:00 Changes: MultiChangesFeed done 
2017-08-22T17:35:56.852+01:00 HTTP+: #003:     --> 200 OK  (0.0 ms)
```